### ISSUE TITLE
How to add support for a new model

### ISSUE THREAD

**Issue by saudiwin** [vincentarelbundock/pymarginaleffects#119]
Hey Vincent, it's me, your favorite annoying American programmer!! :sweat_smile:

As I mentioned on Twitter, I just came up with an MLE implementation of ordbetareg using scipy. Estimation of marginal effects with numerical differentiation should be straightforward as it's a standard MLE fit with Hessian, etc. However, there does not yet seem to a be a protocol for adding support for models to the pymarginaleffects package as there is with the R version. Any particular way you want me to go about doing this? I'm not an expert python programmer by any means, but chatGPT has proven to be remarkably helpful...

**Comment by vincentarelbundock** [comment]
Cool cool. I think the best place to look for an example might be here: 

https://github.com/vincentarelbundock/pymarginaleffects/blob/main/marginaleffects/model_pyfixest.py

**Comment by saudiwin** [comment]
That looks great.

Would you want to define that as a general model class? I.e. if class = MarginalEffectsModel and all the methods are implemented, then it could pass the model sanity check.

Practically should I do a PR for this? I think it has to be within the codebase as I can't get around the model check otherwise.

**Comment by vincentarelbundock** [comment]
Yes, it would need to be a class like `ModelOrdBetaReg`, which inherits from `ModelAbstract`, as in the example I showed you.

A PR would be great, because otherwise I probably won't have time to implement this in the near future.

**Comment by saudiwin** [comment]
Yes indeed it does inherit from ModelAbstract, doesn't it 😂. I told you my
Python skills are ... exceptional.

I'll put together a model_ordbetareg.py file and put it in a PR. Then I
will constantly bug you / send threatening emails until you support the
model.

Just kidding! There's no rush. I don't even like Python anyway.
